src/Cli/func/Helpers/PythonHelpers.cs

```diff
@@ -500,8 +500,20 @@ private static async Task RestorePythonRequirementsDocker(string functionAppRoot
                 dockerImage = await ChoosePythonBuildEnvImage();
             }
 
-            if (string.IsNullOrEmpty(dockerSkipPullFlagSetting) ||
-                !(dockerSkipPullFlagSetting.Equals("true", StringComparison.OrdinalIgnoreCase) || dockerSkipPullFlagSetting == "1"))
+            if (dockerImage == Constants.DockerImages.LinuxPython313ImageAmd64)
+            {
+                // creating temp folder for Dockerfile
+                string tempDir = Path.Combine(Path.GetTempPath(), "python313-docker");
```

Comment:
Code in this block and be broken out into its own method i.e. BuildLocalDockerImage(imageName)

And find a way to connect image name to the static resource.

Code in here should be generic for any local build docker image and not just for `DockerfilePython313buildenv`